### DISCUSSION TITLE
Outreachy - Mar 2021

### DISCUSSION THREAD

**Discussion by jpkrohling** [open-telemetry/community#628] [Community Discussions]
The initial applications for the next round of the [Outreachy](https://outreachy.org) internship will open next Monday, February 1. 

I've typically been highly involved in the program, where our project is named "CNCF Tracing" and includes Jaeger and OpenTracing, and more recently, OpenTelemetry. We typically have funding from the CNCF for one intern and I believe it won't be different this time (right, @caniszczyk?)

Unfortunately, I don’t have the time right now to be a mentor for this round, but perhaps you have a project idea that would be a good fit for an internship? I’m more than happy to guide you through and I’m OK in being the community coordinator. 

If you are interested, just ping me wherever you can find me ([github](https://github.com/jpkrohling/), [twitter](https://twitter.com/jpkrohling), gitter, …)

**Comment by alolita** [comment]
hi @jpkrohling I am interested in helping as a mentor. Let me know if I need to suggest the project idea too :-)

**Reply by jpkrohling** [reply]
It would be optimal if you'd bring your ideas, as those would likely be on parts that matter to you. Otherwise, you can take a look at the previous ideas. If none fits, we can collect some new ideas here. 

For instance, how about some performance tests between the Java API+SDK vs. OpenTracing/OpenCensus? This could build on top of the work done by @gsoria during a previous internship.

**Reply by alolita** [reply]
@jpkrohling I have a few proposals that I share with you :-) 
When do I need to share this info by? 
Can you also please share a link for the submission process?

**Reply by gsoria** [reply]
hi @alolita @jpkrohling, I'm more than happy to help if you decide to continue the work from my previous internship. 
I aspire to become an Outreachy mentor someday :)

**Reply by jpkrohling** [reply]
@alolita, I typically collect ideas in a Google Doc or gist, like this: https://gist.github.com/jpkrohling/d4b483f6c6d592215fbb2e837c2da979. According to their schedule, we need to have our project ideas recorded there at most by March 1, 2021 at 4 pm UTC: https://www.outreachy.org/communities/cfp/. Note that the folks applying for the internship will get their initial application status shared by Feb 22, so it would be good to be already listed there by then. If you confirm that you are willing to be a mentor, I'll formally request the funding from the CNCF. Here's a mentor guide: https://www.outreachy.org/mentor/#mentor

@gsoria if we have a project following up on your internship, would you be willing to be a co-mentor for this round? I might indeed have time to be a co-mentor.

**Reply by gsoria** [reply]
@jpkrohling yes, I gladly would. Let me know what the community decides.

**Reply by alolita** [reply]
@jpkrohling thanks for the example - am working on a Google Doc. Noted deadline is Feb 22. Confirming I'm interested in being a OTel mentor for Outreachy.

**Reply by jpkrohling** [reply]
Wonderful, I just requested the budget.

**Reply by jpkrohling** [reply]
I just got the budget approved for up to 2 interns! It would be awesome if we could get two mentors this round :-)

**Reply by mujjazi** [reply]
@jpkrohling I am interested in participating as an intern, please let me know the relevant details required from my side.
I have an experience working with publishing GitHub actions using bash plus scripting experience with JavaScript and Python.
I am proficient with Java having developed several automation frameworks using Java.

**Reply by jpkrohling** [reply]
@mujjazi thanks for the interest. Please, do take a look at outreachy.org for instructions on how to apply to an internship.

**Reply by mujjazi** [reply]
Thanks, I have applied for it

**Comment by jpkrohling** [comment]
@alolita, do you have the project description already? I have another idea for the Jaeger project as well, and I think I might end up having time to be a mentor as well.

The idea that I have recorded so far is this one:

---
# Performance comparison between OpenTelemetry and OpenTracing instrumentation libraries
Mentor: Gaby Soria (@gsoria)
Co-mentor: Juraci Paixão Kröhling (@jpkrohling)
Status: Pending

In a previous internship, Gaby Soria measured the performance costs of using OpenTracing APIs to instrument an application. Since then, the OpenTelemetry project has surged, replacing the OpenTracing project.

Your role in this internship is to reuse Gaby’s work, measuring the performance of the OpenTelemetry libraries, comparing them with the previous results. In the second phase, you’ll add more cases, covering both APIs.

For this project, you’ll need to be proficient with Java, and you’ll leave this internship with a great understanding of benchmarking applications and writing reliable performance tests. You’ll also have great exposure to the OpenTelemetry API.

---

**Reply by jpkrohling** [reply]
@gsoria would you be able to be a mentor for this project? If so, I'll add it today to the CNCF Tracing community at Outreachy's website.

**Reply by gsoria** [reply]
Yes :)

**Reply by jpkrohling** [reply]
The project is there, you'll need to apply to be a co-mentor, which I'll approve right away.

https://www.outreachy.org/outreachy-may-2021-internship-round/communities/cncf-tracing/performance-comparison-between-opentelemetry-and-o/cfp/
